components/channel_invite_modal/channel_invite_modal.tsx

```diff
@@ -256,24 +257,23 @@ export default class ChannelInviteModal extends React.PureComponent<Props, State
                     size='md'
                     username={option.username}
                 />
-
                 <div className='more-modal__details'>
                     <div className='more-modal__name'>
                         <span>
-                            {getLongDisplayNameParts(option).displayName}
+                            {displayUsername(option, this.props.teammateNameDisplaySetting)}
```

Comment:
There are a few conditional cases in the `displayUsername` function in which a username is returned rather than a nickname or first and last name. We should ensure that we can't get into a state where the UI displays two usernames beside each other. That would look odd.

https://github.com/mattermost/mattermost-webapp/blob/2c00894680d6e25912c588cbc2d8a992d50a8308/packages/mattermost-redux/src/utils/user_utils.ts#L33-L37

Reply:
I get what you mean. Right now the behaviour matches the current members list, which will show two usernames beside each other in these cases:
<img src="https://user-images.githubusercontent.com/112951043/190233645-f5043b9a-5851-47fe-8f0f-5d3444b11cfa.png" width=400 alt="">
Which, I agree, looks odd. Should we change the behaviour here too?

Some ideas:
1. If the preference is set to show usernames, should we just omit the second instance of the username? Or, we can show the full name or nickname as the grey/secondary text. Example:
<img src="https://user-images.githubusercontent.com/112951043/190236695-ba053e8e-04e1-4fa6-9d5d-f624091231e4.png" width=400 alt="">
2. If the preference is *not* set to show usernames, but a given user doesn't have a full name or nickname, then the display name must still fallback to the username. Should we omit the second instance of the username here? Example:
<img src="https://user-images.githubusercontent.com/112951043/190235486-438b780a-d427-44b4-8953-46ceb7d1c95d.png" width=400 alt="">

Reply:
> Should we omit the second instance of the username here?

Yes, we should omit the second instance of username and show as you have it in the last screenshot

Reply:
> In this commit the behaviour matches the current members list, which will show two usernames beside each other in these cases:

I think some of these cases may have been missed in the work done for the members list in the RHS. I agree we should try to be consistent. I'm positive there are other places where this inconsistency may also exist so we may need to do some follow-up PRs to make everything consistent. Off the top of my head, the @mention autocomplete, the from: modifier autocomplete in the search and the channel switcher (command-k) modal are a few other places where similar user listings exist